File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/Drunk.lua
function GetName()
    return "Drunk"
end

function GetType()
  return Debuff
end

function GetHookPoints()
  return { Attacking }
end

function OnAttachment(char)
  char:SetVar("drunk_turnCounter", 3)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == Attacking then
    local drunkCounter = char:GetVar("drunk_turnCounter")

    if drunkCounter == 0 then
      this:RemoveFrom(char)
      return
    end

    drunkCounter = drunkCounter + 1

    if char:GetRace() == "Beerman" then
      param1 = param1 * 1.25
    elseif Roll(0, 99) < 25 then
      param1 = 0
    end

    char:SetVar("drunk_turnCounter", drunkCounter)
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/Drunk.cs
using ProjectR.Interfaces;
using ProjectR.Interfaces.Helper;
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting
{
    public class Drunk : Affliction
    {
        public override string Name { get { return "Drunk"; } }

        public override AfflictionType Type { get { return AfflictionType.Debuff; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.Attacking, 
                };
            }
        }

        protected override void OnAttachment(ICharacter character)
        {
            SetVar(character, "drunk_turnCounter", 3);
        }

        protected override void OnAttacking(ICharacter attacker, ICharacter target, ISpell spell, ref double damage, ref double modifier)
        {
            var drunkCounter = GetVar(attacker, "drunk_turnCounter") - 1;
            SetVar(attacker, "drunk_turnCounter", drunkCounter);

            if (attacker.Race == "Beerman")
            {
                damage *= 1.25d;
            }
            else if (RHelper.RollPercentage(25))
            {
                damage = 0d;
            }

            if (drunkCounter > 0)
            {
                return;
            }

            RemoveFrom(attacker);
        }
    }
}